lib/fw.c

```diff
@@ -151,16 +145,22 @@ int switchtec_fw_dlstatus(struct switchtec_dev *dev,
 }
 
 /**
- * @brief Wait for a firmware download chunk to complete
+ * @brief Get firmware download status
  * @param[in]  dev      Switchtec device handle
  * @param[out] status   The current download status
+ * @param[out] bgstatus The current MRPC background status
  * @return 0 on success, error code on failure
- *
- * Polls the firmware download status waiting until it no longer
- * indicates it's INPROGRESS. Sleeps 5ms between each poll.
  */
-int switchtec_fw_wait(struct switchtec_dev *dev,
-		      enum switchtec_fw_dlstatus *status)
+int switchtec_fw_dlstatus(struct switchtec_dev *dev,
+			  enum switchtec_fw_dlstatus *status,
+			  enum mrpc_bg_status *bgstatus)
+{
+	return __switchtec_fw_dlstatus(dev, MRPC_FWDNLD, status, bgstatus);
+}
+
+static int __switchtec_fw_wait(struct switchtec_dev *dev,
+			       enum mrpc_cmd rpc_cmd,
```

Comment:
It looks a bit odd to use 'rpc' in this context, as we use 'mrpc' more, maybe cmd is good enough.